### PR TITLE
sovrin_register_wallet_type

### PR DESCRIPTION
**Pull request by vimmerru**
Added draft implementation of sovrin_register_wallet_type

### CODE REVIEW

File: src/api/mod.rs [hunk 1]
```diff
@@ -44,6 +44,15 @@ pub enum ErrorCode
     // Caller passed invalid value as param 9 (null, invalid json and etc..)
     CommonInvalidParam9,
 
+    // Caller passed invalid value as param 10 (null, invalid json and etc..)
+    CommonInvalidParam10,
+
+    // Caller passed invalid value as param 11 (null, invalid json and etc..)
+    CommonInvalidParam11,
```
Comment (jovfer) [inline src/api/mod.rs]: Add to headers CommonInvalidParam11 and CommonInvalidParam12

File: src/api/wallet.rs [hunk 2]
```diff
@@ -13,45 +13,82 @@ use self::libc::c_char;
...
+                                                                 config: *const c_char,
+                                                                 runtime_config: *const c_char,
+                                                                 credentials: *const c_char,
+                                                                 handle: *mut i32) -> ErrorCode>,
+                                          set: Option<extern fn(handle: i32,
+                                                                key: *const c_char,
+                                                                value: *const c_char) -> ErrorCode>,
+                                          get: Option<extern fn(handle: i32,
+                                                                key: *const c_char,
+                                                                value_ptr: *const *mut c_char) -> ErrorCode>,
```
Comment (jovfer) [inline src/api/wallet.rs]: Should be `value_ptr *mut *const c_char` here and below

File: src/commands/wallet.rs [hunk 3]
```diff
@@ -39,31 +69,76 @@ impl WalletCommandExecutor {
...
         match command {
+            WalletCommand::RegisterWalletType(xtype, create, open, set, get,
+                                              get_not_expired, list, close, delete, free, cb) => {
+                info! (target: "wallet_command_executor", "RegisterWalletType command received");
+                self.register_type(&xtype, create, open, set,
+                                   get, get_not_expired, list, close, delete, free, cb);
+            }
             WalletCommand::Create(pool_name, name, xtype, config, credentials, cb) => {
-                info!(target: "wallet_command_executor", "Create command received");
+                info! (target: "wallet_command_executor", "Create command received");
```
Comment (jovfer) [inline src/commands/wallet.rs]: Remove spaces after broken autoformat

File: src/api/mod.rs (hunk 1 above)
Comment (vimmerru) [inline src/api/mod.rs]: Done

File: src/commands/wallet.rs (hunk 3 above)
Comment (vimmerru) [inline src/commands/wallet.rs]: Done

File: src/api/wallet.rs (hunk 2 above)
Comment (vimmerru) [inline src/api/wallet.rs]: Done

File: src/api/mod.rs [hunk 4]
```diff
@@ -72,6 +81,9 @@ pub enum ErrorCode
     // Trying to use wallet with pool that has different name
     WalletIncompatiblePoolError,
 
+    // Trying to open wallet that was opened already
+    WalletAlreadyOpenedError,
```
Comment (jovfer) [inline src/api/mod.rs]: - [ ] Add to C header

Comment (vimmerru) [inline src/api/mod.rs]: Done

**Pull request merged**